master/getting-started/mesos/tutorials/connecting-tasks.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Connecting to Tasks
+---
+
+Calico assigns each mesos task a uniquely addressable IP address.
+In testing, it is easy to test connectivity by making requests directly to this
+IP.
+
+However, when designing a multi-tier application, it is not usually sensible to
+hardcode applications to statically connect to these addresses directly, as they are dynamic
+and often change as the tasks die and are reinstantiated across the cluster.
+
+Instead, it is advisable to program the applications to connect to one another
+using DNS entries which dynamically resolve to the applications' current addresses.
+
+In Mesos, there are several Service Discovery solutions which will can perform
+the resolution to the task's container IP.
+
+### Mesos-DNS
+
+Mesos-DNS is an open source DNS service for Mesos which gathers task information
+from the Mesos Master and uses it to serve DNS entries which resolve to a task's IP.
+Mesos-DNS provides the following DNS entry for tasks launched by Marathon:
+
+```
+<task-name>.marathon.mesos
+```
+
+By default, Mesos-DNS will return the IP address of the _Agent_ that is running the task.
+It can be configure to instead return the task's Container IP by modifying its configuration
```

Comment:
Rather than saying "It can be" I think we should say, when using Calico it needs to be reconfigured to return the tasks Container IP instead.